subsys/net/ip/l2/wifi_mgmt.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2016 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_NET_DEBUG_L2_WIFI_MGMT)
+#define SYS_LOG_DOMAIN "net/wifi_mgmt"
+#define NET_LOG_ENABLED 1
+#endif
+
+#include <errno.h>
+
+#include <net/net_core.h>
+#include <net/net_if.h>
+#include <net/wifi_mgmt.h>
+
+static int wifi_connect(u32_t mgmt_request, struct net_if *iface,
+			void *data, size_t len)
+{
+	struct wifi_connect_req_params *params =
+		(struct wifi_connect_req_params *)data;
+
+	SYS_LOG_DBG("%s %u %u %u %s %u",
+		    params->ssid, params->ssid_length,
+		    params->channel, params->security,
+		    params->psk, params->psk_length);
+
+	if ((params->security > WIFI_SECURITY_TYPE_PSK) ||
+	    (params->psk_length < 8) || (params->psk_length > 64) ||
+	    (params->ssid_length > WIFI_SSID_MAX_LEN) ||
+	    ((params->security == WIFI_SECURITY_TYPE_PSK) &&
+	     (params->psk_length == 0)) ||
+	    ((params->channel != WIFI_CHANNEL_ANY) &&
+	     (params->channel > WIFI_CHANNEL_MAX)) ||
+	    !params->ssid || !params->psk) {
+		return -EINVAL;
+	}
```

Comment:
@tbursztyka , this code failed when I tested by attempting to connect to an SSID with open security.

Should the intended test should be something like this?

```
	if (((params->security >= WIFI_SECURITY_TYPE_PSK) &&
	     (!params->psk || (params->psk_length < 8) || 
	      (params->psk_length > 64))) ||
	    (params->ssid_length > WIFI_SSID_MAX_LEN) ||
	    ((params->channel != WIFI_CHANNEL_ANY) &&
	     (params->channel > WIFI_CHANNEL_MAX)) ||
	    !params->ssid) {
		return -EINVAL;
	}
```

Reply:
yep. I haven't been able to test this yet.
In fact I am still struggling to get winc1500 to connect to anything, this crap always fails authenticating and does not tell why. (formerly connecting with static credentials used to work)

Reply:
https://store.ti.com/CC3220SF-LAUNCHXL-SimpleLink-Wi-Fi-CC3220SF-Wireless-Microcontroller-LaunchPad-Development-Kit-P50638.aspx
:-)

Reply:
:D